### ISSUE TITLE
Wrong exported member from .d.ts file

### ISSUE THREAD

**Issue by ikatyang** [Microsoft/TypeScript#17030]
**TypeScript Version:**  2.4.1 / nightly (2.5.0-dev.20170707)

**Code**

(./a.d.ts)

```ts
declare class some_class {}
export declare function some_function(): some_class;
export {};
```

(./b.d.ts)

```ts
declare class some_class {}
declare function some_function(): some_class;
export {some_function};
```

(./c.d.ts)

```ts
declare class some_class {}
export declare function some_function(): some_class;
```

(./test.ts)

```ts
import * as A from './a';
import * as B from './b';
import * as C from './c';

const a_class = new A.some_class(); // error as expected [TS2339]
const b_class = new B.some_class(); // error as expected [TS2339]
const c_class = new C.some_class(); // should throw error, but passed

// [TS2339]: Property 'some_class' does not exist on type 'typeof "<filename>"'.
```

**Expected behavior:**
`a_class`, `b_class` and `c_class` should throw errors.

**Actual behavior:**
`a_class`, `b_class` throw errors, but no `c_class`.

**NOTE**: It works as expected if rename those `*.d.ts` to `*.ts`.

**Comment by andy-ms** [comment]
Everything in a `.d.ts` file is exported by default, so the `export` keyword is redundant.
If the constructor for `some_class` is not exported, you could declare it as an interface instead.

**Comment by ikatyang** [comment]
@andy-ms 

But if there is an `export {}` existed, then everything works as expected (non-`export` won't be exported).

I think there should be no difference between

```ts
declare function some_function(): some_class;
export {some_function};
```

and 

```ts
export declare function some_function(): some_class;
```

since `export function xxx(): void` is just a short-hand of `function xxx(): void; export {xxx};`

---

Isn't it just forgot to mark files that contains `export` modifier as a module?

**Comment by andy-ms** [comment]
That's unfortunate, but if we changed this, a lot of definitions that leave out `export` would be broken.
There's generally no reason to hide something in a definition file; if it's not part of the public interface, you can simply not include it.

**Comment by ikatyang** [comment]
In my case, `some_class` is a part of public API, but the only way to get `some_class` instance is to generate from `some_function`.

As you mentioned above, it can be declared as an interface, but I think it'd be better to use `class` for its semantic meaning.

So, the only way to solve this problem is to add an `export {};`, and this issue is considered `Won't Fix`?

**Comment by andy-ms** [comment]
I don't think there's anything we can do here since we can't make a bunch of implicitly exported members not-exported without breaking a bunch of code.
